src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.RelativeEncoder;
+import com.spikes2212.command.drivetrains.TankDrivetrain;
+import com.spikes2212.command.drivetrains.smartmotorcontrollerdrivetrain.SparkMaxTankDrivetrain;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.dashboard.ChildNamespace;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import frc.robot.RobotMap;
+
+import java.util.function.Supplier;
+
+public class Drivetrain extends SparkMaxTankDrivetrain {
```

Comment:
Add fields with getters that will be used for path following:
`DifferentialDriveOdometry odometry` (update in periodic according to documentation)
`DifferentialDriveKinematics kinematics`
`RamseteController ramseteController`
`Field2d field2d` (no getter, update in periodic according to documentation)

and methods:
`double getLeftSpeed(),
double getRightSpeed(),
Pose2d getPose2d()`
you need to override `configureLoop()`

Also add `FeedForwardSettings` and `TrapezoidProfileSettings`

Add navX json file and add the AHRS, use `Rotation2d` from the AHRS (`gyro.getRotation2d()`)